commit 95b74a7d0c812fbae3574e0fd590bfc19c9e0b93
Author: Radek Novacek <rnovacek@redhat.com>
Date:   Thu Apr 23 14:24:00 2015 +0200

    Filter out all host when filter_host_uuids is empty

diff --git a/config.py b/config.py
index 88a98bd..4c7ab3b 100644
--- a/config.py
+++ b/config.py
@@ -60,14 +60,14 @@ class Config(object):
         self._rhsm_username = rhsm_username
         self._rhsm_password = rhsm_password
 
-        self.filter_host_uuids = []
-        self.exclude_host_uuids = []
+        self.filter_host_uuids = None
+        self.exclude_host_uuids = None
 
         self.hypervisor_id = 'uuid'
 
         # Optional options for backends
-        self.filter_host_parents = []
-        self.exclude_host_parents = []
+        self.filter_host_parents = None
+        self.exclude_host_parents = None
         self.esx_simplified_vim = True
         self.fake_is_hypervisor = True
         self.fake_file = None
@@ -135,12 +135,12 @@ class Config(object):
         try:
             config.filter_host_uuids = parse_list(parser.get(name, "filter_host_uuids"))
         except NoOptionError:
-            config.filter_host_uuids = []
+            config.filter_host_uuids = None
 
         try:
             config.exclude_host_uuids = parse_list(parser.get(name, "exclude_host_uuids"))
         except NoOptionError:
-            config.exclude_host_uuids = []
+            config.exclude_host_uuids = None
 
         if type == 'esx':
             try:
@@ -150,12 +150,12 @@ class Config(object):
             try:
                 config.filter_host_parents = parse_list(parser.get(name, "filter_host_parents"))
             except NoOptionError:
-                config.filter_host_parents = []
+                config.filter_host_parents = None
 
             try:
                 config.exclude_host_parents = parse_list(parser.get(name, "exclude_host_parents"))
             except NoOptionError:
-                config.exclude_host_parents = []
+                config.exclude_host_parents = None
         elif type == 'fake':
             try:
                 config.fake_is_hypervisor = parser.get(name, "is_hypervisor").lower() not in ("0", "false", "no")
diff --git a/virt/esx/esx.py b/virt/esx/esx.py
index 852d44d..8657e8c 100644
--- a/virt/esx/esx.py
+++ b/virt/esx/esx.py
@@ -134,10 +134,10 @@ class Esx(virt.Virt):
         mapping = {}
         for host_id, host in self.hosts.items():
             parent = host['parent'].value
-            if parent in self.config.exclude_host_parents:
+            if self.config.exclude_host_parents is not None and parent in self.config.exclude_host_parents:
                 self.logger.debug("Skipping host '%s' because its parent '%s' is excluded" % (host_id, parent))
                 continue
-            if len(self.config.filter_host_parents) > 0 and parent not in self.config.filter_host_parents:
+            if self.config.filter_host_parents is not None and parent not in self.config.filter_host_parents:
                 self.logger.debug("Skipping host '%s' because its parent '%s' is not included" % (host_id, parent))
                 continue
 
diff --git a/virt/virt.py b/virt/virt.py
index 6015281..34b3425 100644
--- a/virt/virt.py
+++ b/virt/virt.py
@@ -82,10 +82,10 @@ class HostGuestAssociationReport(AbstractVirtReport):
         assoc = {}
         logger = logging.getLogger("rhsm-app")
         for host, guests in self._assoc.items():
-            if host in self._config.exclude_host_uuids:
+            if self._config.exclude_host_uuids is not None and host in self._config.exclude_host_uuids:
                 logger.debug("Skipping host '%s' because its uuid is excluded" % host)
                 continue
-            if len(self._config.filter_host_uuids) > 0 and host not in self._config.filter_host_uuids:
+            if self._config.filter_host_uuids is not None and host not in self._config.filter_host_uuids:
                 logger.debug("Skipping host '%s' because its uuid is not included" % host)
                 continue
             assoc[host] = guests
